Dockerfile

```diff
@@ -1,20 +1,20 @@
+ARG image_tag=latest
+FROM elifesciences/profiles_venv:${image_tag} as venv
 FROM elifesciences/python:b22fbad8f66100d88cb4bd7c7d092b376a9e09bc
 
-USER elife
 ENV PROJECT_FOLDER=/srv/profiles
-RUN mkdir ${PROJECT_FOLDER}
-WORKDIR /srv/profiles
-COPY --chown=elife:elife install.sh requirements.txt /srv/profiles/
-RUN PROFILES_SKIP_DB=1 /bin/bash install.sh
-
-COPY --chown=elife:elife manage.py /srv/profiles/
-COPY --chown=elife:elife migrations/ /srv/profiles/migrations/
-COPY --chown=elife:elife profiles/ /srv/profiles/profiles/
-COPY --chown=elife:elife smoke_tests_wsgi.sh /srv/profiles/
-RUN mkdir /srv/profiles/var/
+WORKDIR ${PROJECT_FOLDER}
 
 USER root
-RUN mkdir var/logs && chown www-data:www-data var/logs
+RUN mkdir -p var/logs && \
+    chown --recursive elife:elife . && \
+    chown www-data:www-data var/logs
+
+COPY --chown=elife:elife smoke_tests_wsgi.sh ./
+COPY --chown=elife:elife manage.py ./
+COPY --chown=elife:elife migrations/ migrations/
+COPY --from=venv --chown=elife:elife ${PROJECT_FOLDER}/venv/ venv/
```

Comment:
we need to understand under which conditions a virtualenv is portable between different operating system images.

For example, there are binaries not necesaarily identical across architectures and that may be depending on shared objects libraries:
```
[giorgio@Newton:~/code/profiles/venv/bin]$ ls -1 . | grep -v py | xargs file | grep ELF
uwsgi:            ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.6.32, BuildID[sha1]=3ad79a028c941d6c3217ffbbcfffd5781b891926, not stripped
```

Plus there are a lot of symlinks to system-wide files:
```
[giorgio@Newton:~/code/profiles/venv]$ ls -l lib/python3.5/ | head
total 44
lrwxrwxrwx   1 giorgio giorgio    25 Feb 19 15:01 abc.py -> /usr/lib/python3.5/abc.py
lrwxrwxrwx   1 giorgio giorgio    28 Feb 19 15:01 base64.py -> /usr/lib/python3.5/base64.py
lrwxrwxrwx   1 giorgio giorgio    28 Feb 19 15:01 bisect.py -> /usr/lib/python3.5/bisect.py
lrwxrwxrwx   1 giorgio giorgio    33 Feb 19 15:01 _bootlocale.py -> /usr/lib/python3.5/_bootlocale.py
lrwxrwxrwx   1 giorgio giorgio    28 Feb 19 15:01 codecs.py -> /usr/lib/python3.5/codecs.py
lrwxrwxrwx   1 giorgio giorgio    30 Feb 19 15:01 collections -> /usr/lib/python3.5/collections
lrwxrwxrwx   1 giorgio giorgio    38 Feb 19 15:01 _collections_abc.py -> /usr/lib/python3.5/_collections_abc.py
lrwxrwxrwx   1 giorgio giorgio    47 Feb 19 15:01 config-3.5m-x86_64-linux-gnu -> /usr/lib/python3.5/config-3.5m-x86_64-linux-gnu
lrwxrwxrwx   1 giorgio giorgio    26 Feb 19 15:01 copy.py -> /usr/lib/python3.5/copy.py
```

This should all be fine if the base images are the same.

Reply:
Explicit strategy on this is to rely on the base images being identical? It can get out of sync given there is a base image in the middle